worlds/doom_ii/__init__.py

```diff
@@ -0,0 +1,268 @@
+import functools
+import logging
+from typing import Any, Dict, List
+
+from BaseClasses import Entrance, CollectionState, Item, ItemClassification, Location, MultiWorld, Region, Tutorial
+from worlds.AutoWorld import WebWorld, World
+from . import Items, Locations, Maps, Options, Regions, Rules
+
+logger = logging.getLogger("DOOM II")
+
+DOOM_TYPE_LEVEL_COMPLETE = -2
+DOOM_TYPE_COMPUTER_AREA_MAP = 2026
+
+
+class DOOM2Location(Location):
+    game: str = "DOOM II"
+
+
+class DOOM2Item(Item):
+    game: str = "DOOM II"
+
+
+class DOOM2Web(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the DOOM II randomizer connected to an Archipelago Multiworld",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Daivuk"]
+    )]
+    theme = "dirt"
+
+
+class DOOM2World(World):
+    """
+    Doom II, also known as Doom II: Hell on Earth, is a first-person shooter game by id Software.
+    It was released for MS-DOS in 1994.
+    Compared to its predecessor, Doom II features larger levels, new enemies, a new "super shotgun" weapon
+    """
+    option_definitions = Options.options
+    game = "DOOM II"
+    web = DOOM2Web()
+    data_version = 3
+    required_client_version = (0, 3, 9)
+
+    item_name_to_id = {data["name"]: item_id for item_id, data in Items.item_table.items()}
+    item_name_groups = Items.item_name_groups
+
+    location_name_to_id = {data["name"]: loc_id for loc_id, data in Locations.location_table.items()}
+    location_name_groups = Locations.location_name_groups
+
+    starting_level_for_episode: List[str] = [
+        "Entryway (MAP01)",
+        "The Factory (MAP12)",
+        "Nirvana (MAP21)"
+    ]
+
+    # Item ratio that scales depending on episode count. These are the ratio for 3 episode. In DOOM1.
+    # The ratio have been tweaked seem, and feel good.
+    items_ratio: Dict[str, float] = {
+        "Armor": 41,
+        "Mega Armor": 25,
+        "Berserk": 12,
+        "Invulnerability": 10,
+        "Partial invisibility": 18,
+        "Supercharge": 28,
+        "Medikit": 15,
+        "Box of bullets": 13,
+        "Box of rockets": 13,
+        "Box of shotgun shells": 13,
+        "Energy cell pack": 10
+    }
+
+    def __init__(self, world: MultiWorld, player: int):
+        self.included_episodes = [1, 1, 1, 0]
+        self.location_count = 0
+
+        super().__init__(world, player)
```

Comment:
I don't see the value in overriding `__init__`? These two things can just be data members with default values like the others.

Reply:
I need those 2 to be instance members to the world. If I put them above they will be static members.

Reply:
Ah, my mistake. I'm conflating behaviour with other langs.